### PR TITLE
[PHPCS] Empower runner

### LINKED ISSUE #550: [PHP_CodeSniffer] Unify processing a large string
`Processor::CodeSniffer#print_large_string` should be replaced with just `TraceWriter#message`, which splits a large string internally.
This logic is duplicate.

https://github.com/sider/runners/blob/8e54d2e52b6fb34df8ca96e68cfe760fd9112589/lib/runners/processor/code_sniffer.rb#L141-L151

https://github.com/sider/runners/blob/8e54d2e52b6fb34df8ca96e68cfe760fd9112589/lib/runners/trace_writer.rb#L72-L83

### PR DESCRIPTION
**Pull request by ybiquitous**
- Save issue details (type, severity, fixable).
- Improve handling exit codes. It raises an analysis error when not exiting with `0`.
  - See <https://github.com/squizlabs/PHP_CodeSniffer/wiki/Configuration-Options#ignoring-errors-when-generating-the-exit-code>
- Remove duplicate logic for `TraceWriter`. Fix #550.

### CODE REVIEW

**Comment by ybiquitous** [comment]
@yykamei Sorry, yet WIP 🙇

**Comment by ybiquitous** [comment]
@yykamei Could you please review?

File: lib/runners/processor/code_sniffer.rb [hunk 1]
```diff
@@ -140,55 +146,43 @@ def php_framework
...
+        "--runtime-set", "ignore_warnings_on_exit", "1", # See https://github.com/squizlabs/PHP_CodeSniffer/wiki/Configuration-Options#ignoring-warnings-when-generating-the-exit-code
         *options,
         target
       )
 
-      stdout = output.read
-      trace_writer.message "Result Output:"
-      print_large_string(stdout)
+      unless output_file.exist?
+        return Results::Failure.new(guid: guid, analyzer: analyzer, message: "No JSON output.")
```
Comment (yykamei) [inline lib/runners/processor/code_sniffer.rb]: Why do you use `Failure` in this context? The situation that the `output_file` does not exist is the users' fault?
If so, `Failure` is appropriate :smiley:

Comment (ybiquitous) [reply lib/runners/processor/code_sniffer.rb]: It is the following case for example.

```
# phpcs --report=json --report-json=/tmp/d20191203-8-1dsw2qx/output.txt --standard=phpcs.xml --extensions=php
ERROR: the "phpcs.xml" coding standard is not installed. The installed coding standards are PSR1, Squiz, PEAR, PSR12, PSR2, MySource, Zend, CakePHP, Symfony, WordPress-Docs, WordPress-Extra, WordPress-Core and WordPress
exit status: 3
No such file or directory @ rb_sysopen - /tmp/d20191203-8-1dsw2qx/output.txt (Errno::ENOENT)
```

But I'm not sure if there may be other cases.

Comment (yykamei) [reply lib/runners/processor/code_sniffer.rb]: I see :+1: Thank you for showing an example. Yes, I agree, it seems caused by users' misconfiguration. Warnings are output by `TraceWriter`, so users can detect what is happening. It seems Okay.

**Comment by ybiquitous** [comment]
Thank you. I'll merge when CI passes.

**Pull request merged**